pkg/agent/runner.go

```diff
@@ -190,14 +190,18 @@ func (r *Runner) Run(ctx context.Context, logger *zap.Logger, vmInfoUpdated util
 
 	execLogger := logger.Named("exec")
 
+	// Add a small random amount to core.Config.PluginRequestTick so that periodic requests tend to
+	// become distribted randomly over time.
+	pluginRequestJitter := util.NewTimeRange(time.Millisecond, 0, 100).Random()
+
 	coreExecLogger := execLogger.Named("core")
 	executorCore := executor.NewExecutorCore(coreExecLogger, getVmInfo(), executor.Config{
 		OnNextActions: r.global.metrics.runnerNextActions.Inc,
 		Core: core.Config{
 			ComputeUnit:                        r.global.config.Scaling.ComputeUnit,
 			DefaultScalingConfig:               r.global.config.Scaling.DefaultConfig,
 			NeonVMRetryWait:                    time.Second * time.Duration(r.global.config.Scaling.RetryFailedRequestSeconds),
-			PluginRequestTick:                  time.Second * time.Duration(r.global.config.Scheduler.RequestAtLeastEverySeconds),
+			PluginRequestTick:                  time.Second*time.Duration(r.global.config.Scheduler.RequestAtLeastEverySeconds) + pluginRequestJitter,
```

Comment:
nit: maybe subtract here? Because `RequestAtLeastEverySeconds = 1` might not give *at least every second*, but *every 1.1 second*.

Also, do we ideally want to randomize every request delay, not just at the start?

Reply:
Ah, good catch :)

My thinking for modifying `PluginRequestTick` itself (rather than waiting at the start) is twofold:

1. Unlike fetching metrics, we _really do_ want to let the plugin know as soon as possible, because delay forces it to act on uncertain data (this probably isn't a _huge_ issue, but maybe it could be, if a ton of VMs are created all at once)
2. There may still be ways for the requests for each VM to become synchronized after startup, although none come to mind at the moment. (I guess in the future, with neondatabase/cloud#6701 this would be more likely, but for now not as much).

Reply:
Switched to subtraction in bc53fe566f722877ad0e4531aaa58076512782d5